napari/_qt/layer_controls/qt_image_controls.py

```diff
@@ -60,8 +60,8 @@ class QtImageControls(QtBaseImageControls):
 
     layer: 'napari.layers.Image'
 
-    def __init__(self, layer):
-        super().__init__(layer)
```

Comment:
One thing that I'm not sure about is having this as a parameter; is that necessary, or can we just initialize it like everything else and only update it via events?

Reply:
The main reason I did it this way is that if a layer is added to the viewer (i.e. without `ndisplay` changing), we have to assign the correct value of from `QtLayerControlsContainer` to the layer control panel somehow. Passing the value as an initialization parameter means that any other initialization that might be dependent on that value has access to it immediately. It's also more atomic - i.e. we can avoid initializing then setting either explicitly or through manually calling a callback/event.

Reply:
Does that explanation seem good enough? Just wondering before I change `**kwargs` to `*, ndisplay: int = 2` for the other layer types.

Reply:
I actually lean more towards explicitly calling the callback at the end of the init. This is what we do with all the `VispyLayer`s (and soon `Overlays`), and since I'm familiar with that part of the codebase, doing it that way feels more "consistent" to me.

However, since here the controls do not have references to the viewer, we can just do it in the container by adding this line to `_add()`:

```py
class QtLayerControlsContainer(QStackedWidget):
    def _add(self, event):
		 ...
        controls.ndisplay = viewer.ndisplay
		 ...
```

Reply:
I lean the other way, but don't feel too strongly. So in the interest of trying new things and being more consistent with other parts of the code base, I made the changes you suggested.